SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+
+    .PARAMETER DataVolume will hold the volume letter of the Data disc. If left empty, it will default to C, unless there is a drive named like Data
+
+    .PARAMETER LogVolume will hold the volume letter of the Log disc. If left empty, it will default to C, unless there is a drive named like Log
+
+    .PARAMETER TempVolume will hold the volume letter of the Temp disc. If left empty, it will default to C, unless there is a drive named like Temp
+
+    .PARAMETER BackupVolume will hold the volume letter of the Backup disc. If left empty, it will default to C, unless there is a drive named like Backup
+
+    .PARAMETER PerformVolumeMaintenance will set the policy for grant or deny this right to the SQL Server service account.
+
+    .PARAMETER SaveFile will prompt you for a file location to save the new config file. Otherwise it will only be saved in the PowerShell bin directory.
+
+    .PARAMETER Authentication will prompt you if you want mixed mode authentication or just Windows AD authentication. With Mixed Mode, you will be prompted for the SA password.
+
+    .PARAMETER EnableException
+            By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+            This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .Inputs
+    None
+
+    .Outputs
+    None
+
+    .Example
+    C:\PS> Install-DbaInstance
+
+    This will run the installation with the default settings
+
+    .Example
+
+    C:\PS> Install-DbaInstance -AppVolume "G"
+
+    This will run the installation with default setting apart from the application volume, this will be redirected to the G drive.
+
+    .Example 
+
+    C:\PS> Install-DbaInstance -Version 2016 -AppVolume "D" -DataVolume "E" -LogVolume "L" -PerformVolumeMaintenance "Yes" -SqlServerAccount "MyDomain\SvcSqlServer"
+
+    This will install SQL Server 2016 on the D drive, the data on E, the logs on L and the other files on the autodetected drives. The perform volume maintenance
+    right is granted and the domain account SvcSqlServer will be used as the service account for SqlServer.
+
+
+    #>
+    Param  (
+        [ValidateSet("2008", "2008R2", "2012", "2014", "2016", "2017", "2019")][string]$Version, 
+        [ValidateSet("Express", "Standard", "Enterprise", "Developer")][string]$Edition,
+        [ValidateSet("AllFeaturesWithDefaults", "Custom")][string]$Role,
+        [ValidateSet("Python", "R", "Python and R")][string]$StatsAndMl,
+        [string]$AppVolume, 
+        [string]$DataVolume, 
+        [string]$LogVolume, 
+        [string]$TempVolume, 
+        [string]$BackupVolume,
+        [string]$InstallFolder,
+        [ValidateSet("Yes", "No")][string]$PerformVolumeMaintenance,
```

Comment:
Booleans are "switches", which allow you to do `-PerformVolumeMaintenance` for $true, or don't pass the param for false. This would need to be

`[switch]$PerformVolumeMaintenance`